arrow-core-data/src/main/kotlin/arrow/core/Ior.kt

```diff
@@ -134,6 +149,161 @@ sealed class Ior<out A, out B> : IorOf<A, B> {
+    ): Ior<A, E> =
+      mapN(SA, b, c, d, Ior.unit, Ior.unit, Ior.unit, Ior.unit, Ior.unit, Ior.unit, Ior.unit) { b, c, d, _, _, _, _, _, _, _ -> map(b, c, d) }
+
+    inline fun <A, B, C, D, E, F> mapN(
+      SA: Semigroup<A>,
+      b: Ior<A, B>,
+      c: Ior<A, C>,
+      d: Ior<A, D>,
+      e: Ior<A, E>,
+      crossinline map: (B, C, D, E) -> F
+    ): Ior<A, F> =
+      mapN(SA, b, c, d, e, Ior.unit, Ior.unit, Ior.unit, Ior.unit, Ior.unit, Ior.unit) { b, c, d, e, _, _, _, _, _, _ -> map(b, c, d, e) }
+
+    inline fun <A, B, C, D, E, F, G> mapN(
+      SA: Semigroup<A>,
+      b: Ior<A, B>,
+      c: Ior<A, C>,
+      d: Ior<A, D>,
+      e: Ior<A, E>,
+      f: Ior<A, F>,
+      crossinline map: (B, C, D, E, F) -> G
+    ): Ior<A, G> =
+      mapN(SA, b, c, d, e, f, Ior.unit, Ior.unit, Ior.unit, Ior.unit, Ior.unit) { b, c, d, e, f, _, _, _, _, _ -> map(b, c, d, e, f) }
+
+    inline fun <A, B, C, D, E, F, G, H> mapN(
+      SA: Semigroup<A>,
+      b: Ior<A, B>,
+      c: Ior<A, C>,
+      d: Ior<A, D>,
+      e: Ior<A, E>,
+      f: Ior<A, F>,
+      g: Ior<A, G>,
+      crossinline map: (B, C, D, E, F, G) -> H
+    ): Ior<A, H> =
+      mapN(SA, b, c, d, e, f, g, Ior.unit, Ior.unit, Ior.unit, Ior.unit) { b, c, d, e, f, g, _, _, _, _ -> map(b, c, d, e, f, g) }
+
+    inline fun <A, B, C, D, E, F, G, H, I> mapN(
+      SA: Semigroup<A>,
+      b: Ior<A, B>,
+      c: Ior<A, C>,
+      d: Ior<A, D>,
+      e: Ior<A, E>,
+      f: Ior<A, F>,
+      g: Ior<A, G>,
+      h: Ior<A, H>,
+      crossinline map: (B, C, D, E, F, G, H) -> I
+    ): Ior<A, I> =
+      mapN(SA, b, c, d, e, f, g, h, Ior.unit, Ior.unit, Ior.unit) { b, c, d, e, f, g, h, _, _, _ -> map(b, c, d, e, f, g, h) }
+
+    inline fun <A, B, C, D, E, F, G, H, I, J> mapN(
+      SA: Semigroup<A>,
+      b: Ior<A, B>,
+      c: Ior<A, C>,
+      d: Ior<A, D>,
+      e: Ior<A, E>,
+      f: Ior<A, F>,
+      g: Ior<A, G>,
+      h: Ior<A, H>,
+      i: Ior<A, I>,
+      crossinline map: (B, C, D, E, F, G, H, I) -> J
+    ): Ior<A, J> =
+      mapN(SA, b, c, d, e, f, g, h, i, Ior.unit, Ior.unit) { b, c, d, e, f, g, h, i, _, _ -> map(b, c, d, e, f, g, h, i) }
+
+    inline fun <A, B, C, D, E, F, G, H, I, J, K> mapN(
+      SA: Semigroup<A>,
+      b: Ior<A, B>,
+      c: Ior<A, C>,
+      d: Ior<A, D>,
+      e: Ior<A, E>,
+      f: Ior<A, F>,
+      g: Ior<A, G>,
+      h: Ior<A, H>,
+      i: Ior<A, I>,
+      j: Ior<A, J>,
+      crossinline map: (B, C, D, E, F, G, H, I, J) -> K
+    ): Ior<A, K> =
+      mapN(SA, b, c, d, e, f, g, h, i, j, Ior.unit) { b, c, d, e, f, g, h, i, j, _ -> map(b, c, d, e, f, g, h, i, j) }
+
+    inline fun <A, B, C, D, E, F, G, H, I, J, K, L> mapN(
+      SA: Semigroup<A>,
+      b: Ior<A, B>,
+      c: Ior<A, C>,
+      d: Ior<A, D>,
+      e: Ior<A, E>,
+      f: Ior<A, F>,
+      g: Ior<A, G>,
+      h: Ior<A, H>,
+      i: Ior<A, I>,
+      j: Ior<A, J>,
+      k: Ior<A, K>,
+      crossinline map: (B, C, D, E, F, G, H, I, J, K) -> L
```

Comment:
Actually, I think with this implementation we still can't call `suspend` functions inside can we?